carbonmark/src/Carbonmark.ts

```diff
@@ -21,6 +21,15 @@ export function handleListingCreated(event: ListingCreated): void {
   let tokenContract = ERC20.bind(event.params.token)
   let tokenSymbol = tokenContract.try_symbol()
 
+  // If the token is not an ERC20, it is an ERC1155
```

Comment:
Token symbols are technically optional as defined in [ERC-20.](https://eips.ethereum.org/EIPS/eip-20)

From the implementation of [ERC-1155](https://eips.ethereum.org/EIPS/eip-1155)

> Smart contracts implementing the ERC-1155 standard MUST implement the ERC-165 supportsInterface function and MUST return the constant value true if 0xd9b67a26 is passed through the interfaceID argument.

Since this is a requirement for the 1155 token, it might be better to test and validate if it's an 1155, then call/assign values as needed.